src/agent0/ethpy/hyperdrive/interface/_mock_contract.py

```diff
@@ -46,13 +64,25 @@ def _calc_effective_share_reserves(pool_state: PoolState) -> FixedPoint:
     return FixedPoint(scaled_value=int(effective_share_reserves))
 
 
-def _calc_spot_price(pool_state: PoolState):
+def _calc_bonds_given_shares_and_rate(
+    pool_state: PoolState,
+    target_rate: FixedPoint,
+    target_shares: FixedPoint | None = None,
+) -> FixedPoint:
     """See API for documentation."""
-    spot_price = hyperdrivepy.calculate_spot_price(
-        fixedpoint_to_pool_config(pool_state.pool_config),
-        fixedpoint_to_pool_info(pool_state.pool_info),
+    if target_shares is None:
+        target_shares = _calc_effective_share_reserves(pool_state)
+    return FixedPoint(
+        scaled_value=int(
+            hyperdrivepy.calculate_initial_bond_reserves(
+                str(target_shares.scaled_value),
+                str(pool_state.pool_config.initial_vault_share_price.scaled_value),
```

Comment:
should we use current vault share price if we want this function to be used any time other than the beginning of the pool?

Reply:
good idea; I made an issue for it so that it is fixed at the root level https://github.com/delvtech/hyperdrive/issues/987

at that time we can propagate the change up